### PR TITLE
DOC: Move users docs from nipype to nipype_tutorial

### PR DESCRIPTION
**Pull request by miykael**
## Summary

PR https://github.com/miykael/nipype_tutorial/pull/77 and https://github.com/miykael/nipype_tutorial/pull/128 moved the content under [User Guide](https://nipype.readthedocs.io/en/1.1.2/users/index.html) already to the `nipype_tutorial`, but we haven't removed this content from `nipype`, yet. This PR takes care of that.

## Deviation between the content

There are three sections that have been copied to the tutorial but either leads to errors or are still WIP:

#### [Saving Workflows and Nodes to a file (experimental)](https://miykael.github.io/nipype_tutorial/notebooks/wip_saving_workflows.html)
Running `wf.export('special_workflow.py')` in the notebook will lead to a `TypeError: a bytes-like object is required, not 'str'`. As the content of this section is still "experimental", I think it's ok if we leave it as WIP.

#### [Running Nipype Interfaces from the command line (nipype_cmd)](https://miykael.github.io/nipype_tutorial/notebooks/wip_nipype_cmd.html)
The content between `nipype` and `nipype_tutorial` is identical. But currently, we're not running any tests, as the command `nipype_cmd` doesn't work inside the docker image. I haven't used `nipype_cmd` yet myself, but it seems `nipype` needs to be installed via apt install python-nipype for this to work?

#### [Resource Scheduling and Profiling with Nipype](https://miykael.github.io/nipype_tutorial/notebooks/wip_resource_sched_profiler.html)
The content between `nipype` and `nipype_tutorial` is identical, but running the profiler within the docker image leads to the error `AttributeError: 'list' object has no attribute 'startTime'`. This is caused by the fact that the profiling information is not handed over as an object but as an object inside a list.

## To consider

Removing the "User Docs" from `nipype` leads to new issues that need to be considered:

1. `install.rst` wasn't removed from `doc/users` as it is also needed in `nipype`'s [Quickstart](https://nipype.readthedocs.io/en/1.1.2/quickstart.html) section.
2. `doc/users` contains a folder `examples` that contains the workflow examples under [Interfaces, Workflows and Examples](https://nipype.readthedocs.io/en/1.1.2/documentation.html). Once those workflows are included under `niflows`, this folder can be removed.
3. This PR inverts the order between the "Developer Guides" and the "Interfaces, Workflows and Examples" section, to this: 
new_documentation_layout
4. This PR puts the `API` and `Developer Guide` into two columns (see below). But the current setup has titles, as well as links to the corresponding sections. Should we take care of this?
developer_title

### CODE REVIEW

File: doc/users/install.rst [hunk 1]
```diff
@@ -16,7 +16,8 @@ image from Docker hub::
     docker pull nipype/nipype
 
 You may also build custom docker containers with specific versions of software
-using Neurodocker_ (see the :doc:`neurodocker`).
+using Neurodocker_ (see the `Neurodocker tutorial
+<https://miykael.github.io/nipype_tutorial/notebooks/introduction_neurodocker.html>`_).
```
Comment (effigies) [inline doc/users/install.rst]: Can we move this to `nipype.github.io/tutorial`? Not sure if we already had discussion there...

Comment (miykael) [reply doc/users/install.rst]: You're talking about the whole `nipype_tutorial`? Fine with me, but we haven't discussed it yet, or decided what to do.

Comment (effigies) [reply doc/users/install.rst]: Yeah.

**Comment by effigies** [comment]
This seems fine to me. Are there any outstanding issues? Can you merge master to get the latest version of testing in?

One quick note: `nipype_cmd` is now `nipypecli`. Not sure if that makes a difference for your notebook...

**Comment by effigies** [comment]
Actually, would you mind pushing a build of the docs to your `gh-pages`, so we can check them out on https://miykael.github.io/nipype?

**Comment by miykael** [comment]
I merged with master and renamed `nipype_cmd` to `nipypecli`. The newest version of the docs can be found on my `gh-pages` (https://miykael.github.io/nipype).

### 3 outstanding issues

#### Command line interface
If `nipype_cmd` is the same as `nipypecli`, does this mean that [this](https://nipype.readthedocs.io/en/latest/users/nipypecmd.html) content can be deleted?

#### Saving Workflows and Nodes to a file (experimental)
As mentioned above, [this content](https://nipype.readthedocs.io/en/latest/users/saving_workflows.html) is only in a [WIP notebook](https://github.com/miykael/nipype_tutorial/blob/master/notebooks/wip_saving_workflows.ipynb) and therefore not visible on the tutorial homepage. The problem is that `wf.export('special_workflow.py')` currently leads to an error in the notebook. Should we leave this as a WIP or remove the content?

#### Resource Scheduling and Profiling with Nipype
The [resource profiler content](https://nipype.readthedocs.io/en/latest/users/resource_sched_profiler.html) is only as a [WIP in the tutorial](https://github.com/miykael/nipype_tutorial/blob/master/notebooks/wip_resource_sched_profiler.ipynb) and doesn't work in the notebook, yet. I think this is addressed in https://github.com/nipy/nipype/issues/2571, but the proposed solution doesn't work in the container.

### Mishap
Also, I unfortunately first updated the `gh-pages` from `nipy/nipype`. I didn't had a gh-pages branch yet, and so `git checkout gh-pages` put me directly on the `upstream` one... I'm really sorry for that and am happy to correct it, if necessary (`gh-pages` doesn't seem to be used anymore?). How can I revert my changes without bringing github down (again?)?

**Comment by effigies** [comment]
I reverted `gh-pages` to Satra's latest commit. Fortunately you didn't destroy the history.

Yes, `nipype_cmd` no longer exists. I don't know whether that information can be (usefully) updated for `nipypecli` or if it just needs deleting. I'm content to leave that decision to you.

I would leave the WIP notebooks as WIP at least until 2.0. Perhaps someone will fix the issues? I would go ahead and open issues for those notebooks here, as the fixes probably need to be done here.

Happy to be overruled by @satra on any and all points.

**Comment by effigies** [comment]
@miykael Are you happy with this at least for this release? We can open issues for any remaining things to fix in 1.1.5.

**Comment by miykael** [comment]
@effigies, yes. In my opinion it's good to go. Will keep track about any issues and open missing issues for the WIP on the tutrial homepage.

**Pull request merged**